### PR TITLE
chromium: Add bisect-feature.py for helping analysis

### PR DESCRIPTION
**Pull request by legendlee1314**
Also add one option to enable bisect in chromium.py.

### CODE REVIEW

File: python/chromium.py [hunk 1]
```diff
@@ -1541,10 +1542,13 @@ def _test_run_device(index_device, results):
...
     dir_test_device_product = dir_test_timestamp + '-' + device_product
 
+    if args.test_for_bisect:
+        dir_test_device_product = '{0}/bisect/{1}'.format(dir_test, rev)
+
     connect_device(device_id=device_id)
 
     if not os.path.exists(dir_test_device_product):
-        os.mkdir(dir_test_device_product)
+        os.makedirs(dir_test_device_product)
```
Comment (gyagp) [inline python/chromium.py]: Use ensure_dir() here

File: python/chromium/feature-test-bisect.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+import os
```
Comment (gyagp) [inline python/chromium/feature-test-bisect.py]: No need to import this

File: python/chromium/feature-test-bisect.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+import sys
+
+import shutil
+
+sys.path.append(sys.path[0] + '/..')
+from util import *
+
+def handle_option():
```
Comment (gyagp) [inline python/chromium/feature-test-bisect.py]: Unify the name to parse_arg()

File: python/chromium/feature-test-bisect.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    global dir_chromium, dir_test_bisect, rev_min, rev_max
+    if not os.path.exists(dir_chromium):
+        error(dir_chromium + ' does not exist')
+    backup_dir(dir_chromium)
+
+    # Run by iteration.
+    # import math
+    # print math.log(256, 2) #  8, normal
+    # print math.log(300000, 2) #  18, current max
+    bisect_time, bisect_max_time = 1, 18
```
Comment (gyagp) [inline python/chromium/feature-test-bisect.py]: should be "times", instead of "time".

**Comment by legendlee1314** [comment]
Updated.

File: python/chromium/feature-test-bisect.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+    return 'Fail'
+
+
+def bisect():
+    '''
+        'goal' == 'progression':  min -> Fail, max -> Pass
+        'goal' == 'regression' :  min -> Pass, max -> Fail
+    '''
+    global dir_chromium, dir_test_bisect, rev_min, rev_max, timestamp
```
Comment (gyagp) [inline python/chromium/feature-test-bisect.py]: If you don't need to change them, you don't need to declare them to be global.

File: python/chromium/feature-test-bisect.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+    if not os.path.exists(dir_chromium):
+        error(dir_chromium + ' does not exist')
```
Comment (gyagp) [inline python/chromium/feature-test-bisect.py]: It's better to check this in setup()

File: python/chromium/feature-test-bisect.py [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        else:  # 'Fail'
+            if goal == 'progression':
+                rev_min = rev
+            else:  # 'regression'
+                rev_max = rev
+
+        bisect_times += 1
+
+
+def _setup_device():
```
Comment (gyagp) [inline python/chromium/feature-test-bisect.py]: Seems to be used only once. You actually don't need a function here.

File: python/chromium/feature-test-bisect.py [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+    if not os.path.exists(dir_chromium):
+        error(dir_chromium + ' does not exist')
+    backup_dir(dir_chromium)
+
+    # Run by iteration.
+    # import math
+    # print math.log(256, 2) #  8 for 256 commits, normal
+    # print math.log(300000, 2) #  18 for 300000 commits, current max
+    bisect_times, bisect_max_times = 1, 18
+    while bisect_times < bisect_max_times:
```
Comment (gyagp) [inline python/chromium/feature-test-bisect.py]: I really don't think these variables are necessary.

**Comment by gyagp** [comment]
How about changing the name to bisect-feature.py?

**Comment by legendlee1314** [comment]
Sounds good. Changed to it.

**Pull request merged**